com.sap.cloud.lm.sl.persistence/src/main/java/com/sap/cloud/lm/sl/persistence/services/FileSystemFileService.java

```diff
@@ -115,53 +115,44 @@ private Path getFilesDirectory(String space) throws IOException {
     }
 
     @Override
-    public int deleteAllByFileIds(Map<String, List<String>> spaceToFileIds) throws FileStorageException {
-        int deletedFileAttributes = super.deleteAllByFileIds(spaceToFileIds);
+    public int deleteByModificationTime(Date modificationTime) throws FileStorageException {
+        int deletedFileAttributes = super.deleteByModificationTime(modificationTime);
         LOGGER.debug(MessageFormat.format(Messages.DELETED_FILE_ATTRIBUTES_COUNT, deletedFileAttributes));
-        int deletedFiles = 0;
-        for (String space : spaceToFileIds.keySet()) {
-            for (String fileId : spaceToFileIds.get(space)) {
-                deleteFileContent(space, fileId);
-                deletedFiles++;
-            }
-        }
-        return deletedFiles;
-    }
-
-    private Path getFilesPerSpaceDirectory(String space) {
-        Path filesPerSpaceDirectory = Paths.get(storagePath, space, DEFAULT_FILES_STORAGE_PATH);
-        return filesPerSpaceDirectory;
-    }
-
-    @Override
-    public List<FileEntry> listFiles(String space, String namespace) throws FileStorageException {
-        List<FileEntry> allEntries = super.listFiles(space, namespace);
-        return deleteOrphanedFileAttributes(allEntries);
-    }
-
-    @Override
-    public List<FileEntry> listByModificationTime(final Date modificationTime) throws FileStorageException {
-        List<FileEntry> fileEntriesAttributes = super.listByModificationTime(modificationTime);
-        deleteOrphanedFileAttributes(fileEntriesAttributes);
 
-        final List<FileEntry> oldEntries = new ArrayList<FileEntry>();
+        AtomicInteger deletedFiles = new AtomicInteger();

```

Comment:
Why AtomicInteger? Leave a comment here if it is really needed.

Reply:
It is used because of its usage in inner class SimpleFileVisitor.
Inner classes require variables of enclosing scope to be final or effectively final.
That is why standard "int" will not help whereas AtomicInteger has final reference and its value could be changed.